superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/DependencyList.tsx

```diff
@@ -0,0 +1,202 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useState } from 'react';
+import { styled, t } from '@superset-ui/core';
+import Icons from 'src/components/Icons';
+import { Select } from 'src/components';
+import { CollapsibleControl } from './CollapsibleControl';
+
+interface DependencyListProps {
+  availableFilters: { label: string; value: string }[];
+  dependencies: string[];
+  onDependenciesChange: (dependencies: string[]) => void;
+}
+
+const MainPanel = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const AddFilter = styled.div`
+  ${({ theme }) => `
+    display: inline-flex;
+    flex-direction: row;
+    align-items: center;
+    cursor: pointer;
+    color: ${theme.colors.primary.base};
+    &:hover {
+      color: ${theme.colors.primary.dark1};
+    }
+  `}
+`;
+
+const DeleteFilter = styled(Icons.Trash)`
+  ${({ theme }) => `
+    cursor: pointer;
+    margin-left: ${theme.gridUnit * 2}px;
+    color: ${theme.colors.grayscale.base};
+    &:hover {
+      color: ${theme.colors.grayscale.dark1};
+    }
+  `}
+`;
+
+const RowPanel = styled.div`
+  ${({ theme }) => `
+    display: flex;
+    width: 220px;
+    flex-direction: row;
+    align-items: center;
+    margin-bottom: ${theme.gridUnit}px;
+  `}
+`;
+
+const Label = styled.div`
+  text-transform: uppercase;
+  font-size: ${({ theme }) => theme.typography.sizes.s}px;
+  color: ${({ theme }) => theme.colors.grayscale.base};
+  margin-bottom: ${({ theme }) => theme.gridUnit}px;
+`;
+
+const Row = ({
+  availableFilters,
+  selection,
+  onChange,
+  onDelete,
+}: {
+  availableFilters: { label: string; value: string }[];
+  selection: string;
+  onChange: (id: string, value: string) => void;
+  onDelete: (id: string) => void;
+}) => (
+  <RowPanel>
+    <Select
+      ariaLabel={t('Limit type')}
+      labelInValue
+      options={availableFilters}
+      onChange={option =>
+        onChange(selection, (option as { value: string }).value)
+      }
+      value={availableFilters.find(e => e.value === selection)}
+    />
+    <DeleteFilter iconSize="xl" onClick={() => onDelete(selection)} />
+  </RowPanel>
+);
+
+const List = ({
+  availableFilters = [],
+  dependencies = [],
+  onDependenciesChange,
+}: DependencyListProps) => {
+  const [rows, setRows] = useState<string[]>(dependencies);
+
+  const updateRows = (newRows: string[]) => {
+    setRows(newRows);
+    onDependenciesChange(newRows);
+  };
+
+  const onAdd = () => {
+    const filter = availableFilters.find(
+      availableFilter => !rows.includes(availableFilter.value),
```

Comment:
Improvement suggestion - don't autoselect a filter that leads to circular dependency (but leave it in the list of availableFilters). For example, if we have filters A, B and C, and A depends on B, when user tries to add dependency in filter B, autoselect C instead of A

Reply:
That would be nice indeed. I'll work on this.

Reply:
I implemented your suggestion.